src/components/Fcrackzip/Fcrackzip.tsx

```diff
@@ -135,118 +149,134 @@ const Fcrackzip = () => {
                 handleProcessData,
                 handleProcessTermination
             );
+
             setPid(result.pid);
-            setOutput(result.output);
+
+            const crackedPasswordRegex = /PASSWORD FOUND\! : (.+)/;
+            const crackedPasswordMatch = result.output.match(crackedPasswordRegex);
+
+            if (crackedPasswordMatch) {
+                const crackedPassword = crackedPasswordMatch[1];
+                setOutput(result.output + `\nCracked Password: ${crackedPassword}`);
+                if (checkedUnzip) {
+                    SaveOutputToTextFile(crackedPassword);
+                }
+            } else {
+                setOutput(result.output);
+            }
         } catch (e: any) {
             setOutput(e.message);
+        } finally {
+            setLoading(false);
         }
     };
 
+    /**
+     * Function to clear output.
+     */
     const clearOutput = useCallback(() => {
         setOutput("");
-    }, [setOutput]);
+    }, []);
 
-    // the disable charset logic is commented out
-    // Please remove it if the future implementation does not need to use it
-    // for now the logic stay as commented
     return (
-        <form onSubmit={form.onSubmit(onSubmit)}>
-            <LoadingOverlay visible={loading} />
-            {loading && (
+        <RenderComponent
+            title={title}
+            description={description_guide}
+            steps={steps}
+            tutorial={tutorial}
+            sourceLink={sourceLink}
+        >
+            {!dependencyAvailable && (
                 <div>
-                    <Button variant="outline" color="red" style={{ zIndex: 1001 }} onClick={handleCancel}>
-                        Cancel
-                    </Button>
+                    <p>Dependency not available. Please install it.</p>
```

Comment:
You should not need this once you get checkAllCommandsAvailability working correctly.